client/src/components/RelatedProducts.jsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from 'react';
+import { RelatedProductsList, LoadingSpinner } from '../components';
+import { randomize } from '../lib';
+
+export function RelatedProducts({ category, productId }) {
+  const [related, setRelated] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const categoryArray = category.split(' ');
+  const firstCategory = categoryArray[0]
+
+  useEffect(() => {
+    const getRelated = async () => {
+      try {
+        const products = await fetch(`/api/products/details/${firstCategory}/${productId}`);
+        if (!products.ok) {
+          throw new Error(`Network response was not OK. Status Code: ${products.status}`);
+        }
+        const productsData = await products.json();
+        setRelated(randomize(productsData));
+      } catch (err) {
+        console.error(err);
```

Comment:
Here, also, display an error message to the user. There are other places this may happen, so check the entire code base.